qdrant_client/embed/model_embedder.py

```diff
@@ -42,11 +43,46 @@ def process(self, items: Iterable[tuple[int, Any]]) -> Iterable[tuple[int, Any]]
 class ModelEmbedder:
     MAX_INTERNAL_BATCH_SIZE = 64
 
-    def __init__(self, parser: Optional[ModelSchemaParser] = None, **kwargs: Any):
+    def __init__(
+        self,
+        parser: Optional[ModelSchemaParser] = None,
+        is_local_mode: bool = False,
+        server_version: Optional[str] = None,
+        **kwargs: Any,
+    ):
         self._batch_accumulator: dict[str, list[INFERENCE_OBJECT_TYPES]] = {}
         self._embed_storage: dict[str, list[NumericVector]] = {}
         self._embed_inspector = InspectorEmbed(parser=parser)
-        self.embedder = Embedder(**kwargs)
+        self._is_builtin_embedder_available = self._check_builtin_embedder_availability(
+            is_local_mode, server_version
+        )
+        self.embedder = (
+            Embedder(**kwargs) if FastEmbedMisc.is_installed() else BuiltinEmbedder(**kwargs)
+        )
```

Comment:
Is there a reason why we don't fail here already instead of in `embed_models{,_strict,_batch}`?